### PR TITLE
Add functions to allow async Vaxis.queryTerminal()

### PR DESCRIPTION
**Pull request by neurocyte**

### CODE REVIEW

File: src/Vaxis.zig [hunk 1]
```diff
@@ -197,9 +209,13 @@ pub fn queryTerminal(self: *Vaxis) !void {
...
     _ = try tty.write(ctlseqs.primary_device_attrs);
     try tty.flush();
+}
 
-    // 1 second timeout
-    std.Thread.Futex.timedWait(&self.query_futex, 0, 1 * std.time.ns_per_s) catch {};
+/// Enable features detected by responses to queryTerminal. This function
+/// is only for use with a custom main loop. Call Vaxis.queryTerminal() if
+/// you are using Loop.run()
+pub fn enableTerminalFeatures(self: *Vaxis) !void {
```
Comment (rockorager) [inline src/Vaxis.zig:217]: Bikeshed: I think we should call this `enableDetectedFeatures`. It's a little more clear that this only applies to *some* features. What do you think?

Comment (neurocyte) [reply src/Vaxis.zig:217]: Sure.

File: src/Vaxis.zig [hunk 2]
```diff
@@ -167,6 +167,18 @@ pub fn exitAltScreen(self: *Vaxis) !void {
 /// capabilities will be delivered to the client and possibly intercepted by
 /// Vaxis to enable features
 pub fn queryTerminal(self: *Vaxis) !void {
+    try self.queryTerminalSend();
+
+    // 1 second timeout
+    std.Thread.Futex.timedWait(&self.query_futex, 0, 1 * std.time.ns_per_s) catch {};
+
+    try self.enableTerminalFeatures();
```
Comment (rockorager) [inline src/Vaxis.zig:175]: This should be fixed to `enableDetectedFeatures` now :)

Comment (neurocyte) [reply src/Vaxis.zig:175]: Oops!

**Pull request merged**